src/libraries/System.IO.Pipes/src/Microsoft/Win32/SafeHandles/SafePipeHandle.Unix.cs

```diff
@@ -8,70 +8,138 @@
 using System.Reflection;
 using System.Runtime.InteropServices;
 using System.Security;
+using System.Threading;
 
 namespace Microsoft.Win32.SafeHandles
 {
     public sealed partial class SafePipeHandle : SafeHandleZeroOrMinusOneIsInvalid
     {
         private const int DefaultInvalidHandle = -1;
+        private static Func<IntPtr, bool, Socket>? s_createSocketForPipe;
 
         // For anonymous pipes, SafePipeHandle.handle is the file descriptor of the pipe, and the
-        // _named* fields remain null. For named pipes, SafePipeHandle.handle is a copy of the file descriptor
-        // extracted from the Socket's SafeHandle, and the _named* fields are the socket and its safe handle.
+        // For named pipes, SafePipeHandle.handle is a copy of the file descriptor
```

Comment:
Something's messed up in the comment here; the previous sentence just ends abruptly.